tests/unit_tests/prompts/test_prompt.py

```diff
@@ -145,6 +145,12 @@ def test_partial() -> None:
     assert new_result == "This is a 3 test."
     result = prompt.format(foo="foo")
     assert result == "This is a foo test."
+    # Test partialing with prompt templates
+    compose_prompt = PromptTemplate(input_variables=["bar"], template="sad and {bar}")
```

Comment:
Complex enough that can be stuck in a separate unit test perhaps. Took a bit of time to figure out what this is doing